### PR TITLE
Switch to argumentless (py3) super().

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/mpl_toolkits/axes_grid1/parasite_axes.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def cla(self):
             ax.cla()
 
         self._get_base_axes_attr("cla")(self)
-        #super(HostAxes, self).cla()
+        #super().cla()
```
Comment (efiring) [inline lib/mpl_toolkits/axes_grid1/parasite_axes.py]: This is probably a good time to delete this commented-out code.

File: lib/mpl_toolkits/axisartist/axis_artist.py [hunk 2]
```diff
@@ -461,7 +460,7 @@ def __init__(self, *kl, **kwargs):
 
         axis_direction = kwargs.pop("axis_direction", "bottom")
         self._axis = kwargs.pop("axis", None)
-        #super(AxisLabel, self).__init__(*kl, **kwargs)
+        #super().__init__(*kl, **kwargs)
```
Comment (efiring) [inline lib/mpl_toolkits/axisartist/axis_artist.py]: Delete?

**Comment by efiring** [comment]
Two similar test failures.  Here's one:
```
lib/mpl_toolkits/axes_grid1/parasite_axes.py:319: in twinx
    ax2.axis["left", "top", "bottom"].set_visible(False)
lib/mpl_toolkits/axes_grid1/mpl_axes.py:33: in __getitem__
    [super().__getitem__(k1) for k1 in k])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.0 = <tuple_iterator object at 0x7ff35c6d6da0>
>   [super().__getitem__(k1) for k1 in k])
E   TypeError: super(type, obj): obj must be an instance or subtype of type
lib/mpl_toolkits/axes_grid1/mpl_axes.py:33: TypeError
```
Looks like `super()` with no args doesn't work inside a list comprehension.

**Comment by anntzer** [comment]
yup, looks like `__class__` is resolved in the wrong closure (sort of makes sense). fixed.

**Comment by tacaswell** [comment]
This makes me happy :smiley:

**Comment by anntzer** [comment]
best edits are the ones that are `sed`dable...

**Pull request merged**